src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php

```diff
@@ -129,13 +129,13 @@ private function addTwigOptions(ArrayNodeDefinition $rootNode)
                 ->end()
                 ->scalarNode('autoescape_service')->defaultNull()->end()
                 ->scalarNode('autoescape_service_method')->defaultNull()->end()
-                ->scalarNode('base_template_class')->example('Twig_Template')->end()
+                ->scalarNode('base_template_class')->example('Twig_Template')->cannotBeEmpty()->end()
                 ->scalarNode('cache')->defaultValue('%kernel.cache_dir%/twig')->end()
                 ->scalarNode('charset')->defaultValue('%kernel.charset%')->end()
-                ->scalarNode('debug')->defaultValue('%kernel.debug%')->end()
-                ->scalarNode('strict_variables')->end()
+                ->booleanNode('debug')->defaultValue('%kernel.debug%')->end()
```

Comment:
Using such string resolved later works because the default value is applied after the validation. FrameworkBundle already does it.

It is true however that the DI extension will still receive the string here, not the resolved value. But we don't care because the DI extension does not care about the value itself. It just sets it in the container (where it will be resolved).
For cases where the value influences the DI extension logic, it is indeed necessary to resolve the value, which can either be done explicitly in the DI extension or by passing the debug flag to the Configuration class (which is what DoctrineBundle does for instance)